src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1185 @@
+    // Check if the statement is an assertion regarding a value that can be
+    // used in a simplification later on.
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof MethodCallExpr) {
+        MethodCallExpr mCall = (MethodCallExpr) exp;
+        // Check if the method call is an assertTrue statement
+        if (mCall.getName().equals("assertTrue")) {
+          // Is an assertTrue statement.
+          List<Expression> mArgs = mCall.getArgs();
+          if (mArgs.size() == 1) {
+            // Retrieve and store the value associated with the
+            // variable in the assertion.
+            Expression mExp = mArgs.get(0);
+            List<Node> children = mExp.getChildrenNodes();
+
+            // Check that the expression only has two nodes.
+            if (children.size() != 2) {
+              return;
+            }
+            String mExpStr = mExp.toString();
+
+            // Take the substring without the first and second expression nodes.
+            String firstExp = children.get(0).toString();
+            int secondExpIndex = mExpStr.lastIndexOf(children.get(1).toString());
+            mExpStr = mExpStr.substring(firstExp.length(), secondExpIndex);
+
+            // Check that the comparison is an equality comparison.
+            if (mExpStr.trim().equals("==")) {
+              String var = children.get(0).toString();
+              String val = children.get(1).toString();
+              primitiveValues.put(var, val);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Return a variable declaration statement that replaces the right hand side by 0, false, or null,
+   * whichever is type correct. Returns null if there are multiple variable declarations in a single
+   * statement, such as {@code int i, j, k;}.
+   *
+   * @param vdExpr variable declaration expression representing the current statement to simplify
+   * @return a {@code Statement} object representing the simplified variable declaration expression
+   *     if the type of the variable is a primitive and a value has been previously calculated.
+   *     Otherwise, {@code null} is returned. Also returns {@code null} if more than one variable is
+   *     declared in the {@code VariableDeclarationExpr}.
+   */
+  private static Statement rhsAssignZeroValue(VariableDeclarationExpr vdExpr) {
+    if (vdExpr.getVars().size() != 1) {
+      // Number of variables declared in this expression is not one.
+      return null;
+    }
+    Type type = vdExpr.getType();
+    if (type instanceof PrimitiveType) {
+      switch (((PrimitiveType) type).getType()) {
+        case Boolean:
+          return rhsAssignWithValue(vdExpr, "false");
+        case Char:
+          return rhsAssignWithValue(vdExpr, "");
+        default:
+          return rhsAssignWithValue(vdExpr, "0");
+      }
+    }
```

Comment:
For wrapped primitives such as `Integer`, try the corresponding zero value as well.
That means that this routine will return a list that may be of length 0, 1, or 2.